relay/abi.go

```diff
@@ -75,79 +41,131 @@ func init() {
 	}
 }
 
-func handleLog(vLog *types.Log, w *wallet.Wallet, q *uploader.Queue, chainID uint64, jackalContract string) {
-	event := struct {
-		Sender common.Address
-		Merkle string
-		Size   uint64
-	}{}
-
-	e, err := eventABI.Unpack("PostedFile", vLog.Data)
-	if err != nil {
-		log.Fatalf("Failed to unpack log data normally: %v", err)
-		return
-	}
-
-	fmt.Println(len(e))
-
-	err = eventABI.UnpackIntoInterface(&event, "PostedFile", vLog.Data)
-	if err != nil {
-		log.Fatalf("Failed to unpack log data into interface: %v", err)
-		return
-	}
-
+func generatePostedFileMsg(w *wallet.Wallet, q *uploader.Queue, chainID uint64, jackalContract string, event PostedFile) (err error) {
 	log.Printf("Event details: %+v", event)
-
 	evmAddress := event.Sender.String()
 
 	merkleRoot, err := hex.DecodeString(event.Merkle)
 	if err != nil {
-		log.Fatalf("Failed to decode merkle: %v", err)
+		log.Printf("Failed to decode merkle: %v", err)
 		return
 	}
 
 	abci, err := w.Client.RPCClient.ABCIInfo(context.Background())
 	if err != nil {
-		log.Fatalf("Failed to query ABCI: %v", err)
+		log.Printf("Failed to query ABCI: %v", err)
 		return
 	}
 
 	log.Printf("Relaying for %s\n", event.Sender.String())
 
 	var hours int64 = 100 * 365 * 24
-
 	merkleBase64 := base64.StdEncoding.EncodeToString(merkleRoot)
-
 	var maxProofs int64 = 3
 	fileSize := int64(event.Size)
+
+	note := make(map[string]any)
+	err = json.Unmarshal([]byte(event.Note), &note)
+	if err != nil && event.Note != "" {
+		log.Printf("Could not unmarshal: %v", err)
+		return
+	}
+
+	note["relayed"] = map[string]any{"chain_id": chainRep(chainID), "for": evmAddress}
+	newNote, err := json.Marshal(note)
+	if err != nil {
+		log.Printf("Failed to add memo: %v", err)
+		return
+	}
+
 	storageMsg := evmTypes.ExecuteMsg{
 		PostFile: &evmTypes.ExecuteMsgPostFile{
 			Merkle:        merkleBase64,
 			FileSize:      fileSize,
 			ProofInterval: 3600,
```

Comment:
Can change this to 7200 which is the new default, not a huge deal though since right now the chain ignores this field.

Reply:
changed